.github/workflows/upload_to_S3.yml

```diff
@@ -0,0 +1,51 @@
+name: Upload to S3
+
+on:
+  push:
+    # branches:
+    #   - main
+    workflow_dispatch:
+      inputs:
+        environment:
+          type: environment
+          description: The environment to deploy to.
+
+jobs:
+  detect-environments:
+    runs-on: ubuntu-latest
+    outputs:
+      environments: ${{ steps.environments.outputs.result }}
+    steps:
+      - uses: actions/github-script@v6
+        id: environments
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: json
+          script: |
+            if (context.payload?.inputs?.environment) return [context.payload?.inputs?.environment];
+            const {data: {environments}} = 
+              await github.request(`GET /repos/${process.env.GITHUB_REPOSITORY}/environments`);
+            return environments.map(e => e.name)
+
+  upload_to_S3:
+    runs-on: ubuntu-latest
+    needs: [detect-environments]
+    strategy:
+      matrix:
+        environment: ${{ fromJSON(needs.detect-environments.outputs.environments) }}
+    environment: ${{ matrix.environment }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.DEPLOY_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DEPLOY_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      # - name: Save to S3
```

Comment:
We're not publishing to S3 anymore with this commented?  Is this intentional?

Reply:
Yes, commenting out the S3 sync lines is intentional for testing purposes, the goal is to merge this into main, and then create a new branch to uncomment the lines.